src/module/functions/mod.rs

```diff
@@ -447,98 +447,112 @@ impl Module {
     ///     builder.func_body().unreachable();
-    ///     builder.local_func(vec![])
     /// });
     /// ```
     ///
+    /// The arguments passed to the original function will be passed to the
+    /// new exported function that was built in your closure.
+    ///
     /// This function returns the function ID of the *new* function, and
     /// removes the existing import that has been replaced (the function will become local).
-    pub fn replace_imported_func<F>(&mut self, fid: FunctionId, fn_builder: F) -> Result<FunctionId>
-    where
-        F: FnOnce((&FuncParams, &FuncResults)) -> Result<LocalFunction>,
-    {
-        // If the function is in the imports, replace it
-        match (self.imports.get_imported_func(fid), self.funcs.get(fid)) {
-            (
-                Some(original_imported_fn),
-                Function {
-                    kind: FunctionKind::Import(ImportedFunction { ty: tid, .. }),
-                    ..
-                },
-            ) => {
-                // Retrieve the params & result types for the imported function
-                let ty = self.types.get(*tid);
-                let (params, results) = (ty.params().to_vec(), ty.results().to_vec());
-
-                // Mutate the existing function, changing it from a FunctionKind::ImportedFunction
-                // to the local function produced by running the provided `fn_builder`
-                let func = self.funcs.get_mut(fid);
-                func.kind = FunctionKind::Local(
-                    fn_builder((&params, &results)).context("import fn builder failed")?,
-                );
-
-                self.imports.delete(original_imported_fn.id());
-
-                Ok(fid)
-            }
-            // The export didn't exist, or the function isn't the kind we expect
-            _ => bail!("cannot replace function [{fid:?}], it is not an imported function"),
+    pub fn replace_imported_func(
+        &mut self,
+        fid: FunctionId,
+        builder_fn: impl FnOnce(&mut InstrSeqBuilder),
+    ) -> Result<FunctionId> {
+        let original_import_id = self
+            .imports
+            .get_imported_func(fid)
+            .map(|import| import.id())
+            .with_context(|| format!("no exported function with ID [{fid:?}]"))?;
+
+        if let Function {
+            kind: FunctionKind::Import(ImportedFunction { ty: tid, .. }),
+            ..
+        } = self.funcs.get(fid)
+        {
+            // Retrieve the params & result types for the imported function
+            let ty = self.types.get(*tid);
+            let (params, results) = (ty.params().to_vec(), ty.results().to_vec());
+
+            // Build the new function
+            let mut builder = FunctionBuilder::new(&mut self.types, &params, &results);
+            let mut new_fn_body = builder.func_body();
+            builder_fn(&mut new_fn_body);
+            let new_func_kind = FunctionKind::Local(builder.local_func(vec![]));
```

Comment:
Are no local Ids being passed here? We should pre-generate a new `localId` for each argument here I think?

Reply:
Yeah this was a bit confusing to me, but I looked back at the previous implementations and saw that they were actually always empty.... So I thought that *might* work here. 

One problem I had here was that there *is* no lcoal func to get args from here right? Am I misunderstanding maybe?

Reply:
Yeah in this line you exactly need to be creating the localIds - because otherwise it won't be possible to read arguments for the created function.

Reply:
Ahhh gotcha, thank you -- I didn't understand what you meant by generating the new IDs *per argument*. In the other case we can just use the args but here we have to *create* them, rather than the person being able to (though they theoretically could themselves if they knew which function to replace). 

Will update this bit too

Reply:
OK so I *think* I have it right here -- https://github.com/rustwasm/walrus/pull/250/commits/af96691e6b286dd8847474aee5e732764eff7bdd -- I'm generating new `LocalId`s to match each param w/ the right internal type.